src/watertap_contrib/reflo/costing/solar/thermal_energy_storage.py

```diff
@@ -0,0 +1,118 @@
+
+    costing = blk.parent_block()
+
+    blk.thermal_energy_storage_cost = pyo.Var(
+        initialize=10.79,
+        units=costing.base_currency / pyo.units.kW,
+        bounds=(0, None),
+        doc="Cost per kW storage",
+    )
+
+    blk.contingency_frac_direct_cost = pyo.Var(
+        initialize=0,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Fraction of direct costs for contingency",
+    )
+
+    blk.indirect_frac_direct_cost = pyo.Var(
+        initialize=0.13,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Fraction of direct costs, including contingency, for indirect costs",
+    )
+
+    blk.fixed_operating_by_capacity = pyo.Var(
+        initialize=66,
+        units=costing.base_currency / (pyo.units.kW * costing.base_period),
+        bounds=(0, None),
+        doc="Fixed operating cost of thermal energy storage per kW capacity",
+    )
+
+
+@register_costing_parameter_block(
+    build_rule=build_tes_cost_param_block,
+    parameter_block_name="tes",
+)
+def cost_tes(blk):
+
+    global_params = blk.costing_package
+    tes_params = blk.costing_package.tes
+    tes = blk.unit_model
+    make_capital_cost_var(blk)
+    make_fixed_operating_cost_var(blk)
+
+    blk.direct_capital_cost = pyo.Var(
+        initialize=1e4,
+        units=blk.config.flowsheet_costing_block.base_currency,
+        bounds=(0, None),
+        doc="Direct cost of thermal energy storage",
+    )
+
+    blk.indirect_capital_cost = pyo.Var(
+        initialize=1e4,
+        units=blk.config.flowsheet_costing_block.base_currency,
+        bounds=(0, None),
+        doc="Indirect costs of thermal energy storage",
+    )
+
+    blk.sales_tax = pyo.Var(
+        initialize=1e2,
+        units=blk.config.flowsheet_costing_block.base_currency,
+        bounds=(0, None),
+        doc="Sales tax for thermal energy storage",
+    )
+
+    blk.direct_cost_constraint = pyo.Constraint(
+        expr=blk.direct_capital_cost
+        == (tes_params.thermal_energy_storage_cost * tes.thermal_energy_capacity)
+        * (1 + tes_params.contingency_frac_direct_cost)
+    )
+
+    blk.indirect_cost_constraint = pyo.Constraint(
+        expr=blk.indirect_capital_cost
+        == blk.direct_capital_cost * tes_params.indirect_frac_direct_cost
+    )
+
+    blk.sales_tax_constraint = pyo.Constraint(
+        expr=blk.sales_tax == blk.direct_capital_cost * global_params.sales_tax_frac
+    )
+
+    blk.capital_cost_constraint = pyo.Constraint(
+        expr=blk.capital_cost
+        == blk.direct_capital_cost + blk.indirect_capital_cost + blk.sales_tax
+    )
+
+    blk.fixed_operating_cost_constraint = pyo.Constraint(
+        expr=blk.fixed_operating_cost
+        == tes_params.fixed_operating_by_capacity
+        * pyo.units.convert(tes.heat_load, to_units=pyo.units.kW)
+    )
+
+    # register the flows, e.g.,:
+    blk.costing_package.cost_flow(
+        tes.electricity,
+        "electricity",
+    )
```

Comment:
Any accounting for the possibility of auxiliary heat and associated costs?

Reply:
That's a great question. @MuktaHardikar would like to hear your thoughts on this.